packages/lesswrong/lib/random.ts

```diff
@@ -48,6 +48,14 @@ export const randomId = (length=ID_LENGTH, randIntCallback?: RandIntCallback) =>
   }
 }
 
+/**
+ * Detects whether the given string could *not* have been generated by the `randomId()` function.
+ */
+export const isNotRandomId = (id: string, length=ID_LENGTH): boolean => {
+  const regex = new RegExp(`^[abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTWXYZ23456789]{${length}}$`);
+  return !regex.test(id);
+}
```

Comment:
Because we now accept clientIds that are generated outside the server, this is used to check for malicious input